File: example/hold.cpp
#include "../include/franka_timeout_handler/robot.h"
#include <iostream>

int _main(int argc, char **argv)
{
    if (argc != 2)
    {
        std::cout << "Usage: ./hold IP" << std::endl;
        return 1;
    }
    
    try
    {
        franka_timeout_handler::Robot robot(argv[1]);
        robot.start(franka_timeout_handler::ControllerType::cartesian);
        const size_t time = 10000;
        size_t late = 0;
        for (size_t i = 0; i < time; i++)
        {
            robot.receive();
            robot.send();
            std::cout << (robot.get_late() ? "Late" : "Not late") << std::endl;
            if (robot.get_late()) late++;
        }
        std::cout << "Total: late " << late << " times of " << time << " (" << 100.0 * late / time << "%)" << std::endl;    
    }
    catch (std::exception &e)
    {
        std::cout << e.what() << std::endl;
        return 1;
    }
    return 0;
}

int main(int argc, char **argv)
{
    return _main(argc, argv);
}

File: example/hold.py
#!/bin/python3

import sys
import os
import franka_timeout_handler
import numpy

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Usage: ./hold.py IP")
    else:
        try:
            robot = franka_timeout_handler.Robot(sys.argv[1])
            robot.start(franka_timeout_handler.ControllerType.cartesian)
            time = 10000
            late = 0
            for i in range(time):
                robot.receive()
                robot.send()
                print("Late" if robot.get_late() else "Not late")
                if robot.get_late(): late = late + 1
            print("Total: late ", late, " times of ", time, " (", 100.0 * late /  time, "%)")
        except:
            print(sys.exc_info()[0])